Build error on android

Hi,

I'm getting the following error when building for android with ionic 4:

```
BUILD FAILED in 9s
/Users/Tobi/ProjectN/platforms/android/gradlew: Command failed with exit code 1 Error output:
/Users/Tobi/ProjectN/platforms/android/app/src/main/java/com/sarriaroman/PhotoViewer/PhotoActivity.java:25: error: cannot find symbol
import com.squareup.picasso.UrlConnectionDownloader;
                           ^
  symbol:   class UrlConnectionDownloader
  location: package com.squareup.picasso
/Users/Tobi/ProjectN/platforms/android/app/src/main/java/com/sarriaroman/PhotoViewer/PhotoActivity.java:197: error: cannot find symbol
                picasso = Picasso.with(PhotoActivity.this);
                                 ^
  symbol:   method with(PhotoActivity)
  location: class Picasso
/Users/Tobi/ProjectN/platforms/android/app/src/main/java/com/sarriaroman/PhotoViewer/PhotoActivity.java:203: error: <anonymous com.sarriaroman.PhotoViewer.PhotoActivity$3> is not abstract and does not override abstract method onError(Exception) in Callback
                    .into(photo, new com.squareup.picasso.Callback() {
                                                                     ^
/Users/Tobi/ProjectN/platforms/android/app/src/main/java/com/sarriaroman/PhotoViewer/PhotoActivity.java:209: error: method does not override or implement a method from a supertype
                        @Override
                        ^
/Users/Tobi/ProjectN/platforms/android/app/src/main/java/com/sarriaroman/PhotoViewer/PhotoActivity.java:227: error: cannot find symbol
                    Picasso picasso = Picasso.with(PhotoActivity.this);
                                             ^
  symbol:   method with(PhotoActivity)
  location: class Picasso
/Users/Tobi/ProjectN/platforms/android/app/src/main/java/com/sarriaroman/PhotoViewer/PhotoActivity.java:231: error: <anonymous com.sarriaroman.PhotoViewer.PhotoActivity$4$1> is not abstract and does not override abstract method onError(Exception) in Callback
                                .into(photo, new com.squareup.picasso.Callback() {
                                                                                 ^
/Users/Tobi/ProjectN/platforms/android/app/src/main/java/com/sarriaroman/PhotoViewer/PhotoActivity.java:237: error: method does not override or implement a method from a supertype
                                    @Override
                                    ^
/Users/Tobi/ProjectN/platforms/android/app/src/main/java/com/sarriaroman/PhotoViewer/PhotoActivity.java:335: error: cannot find symbol
        builder.downloader(new UrlConnectionDownloader(ctx) {
                               ^
  symbol:   class UrlConnectionDownloader
  location: class PhotoActivity
/Users/Tobi/ProjectN/platforms/android/app/src/main/java/com/sarriaroman/PhotoViewer/PhotoActivity.java:336: error: method does not override or implement a method from a supertype
            @Override
            ^
/Users/Tobi/ProjectN/platforms/android/app/src/main/java/com/sarriaroman/PhotoViewer/PhotoActivity.java:338: error: cannot find symbol
                HttpURLConnection connection = super.openConnection(uri);
                                                    ^
  symbol: method openConnection(Uri)
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
10 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 9s
[ERROR] An error occurred while running subprocess cordova.
        
        cordova build android exited with exit code 1.
        
```

I already tried with version 1.1.18 and 1.2.1 but no one works...

Anyone an idea?

--- a/README.md
+++ b/README.md
@@ -116,3 +116,7 @@ PhotoViewer.show('http://my_site.com/my_image.jpg', 'Optional Title', options);
 
 (1.2.4)
 - Removing Source Maps because of errors
+
+(1.2.5) 
+- Update Picasso Library to 2.71828
+- Fixed Picasso Bug

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "com-sarriaroman-photoviewer",
-  "version": "1.2.4",
+  "version": "1.2.5",
   "description": "This plugin is intended to show a picture from an URL into a Photo Viewer with zoom features.",
   "cordova": {
     "id": "com-sarriaroman-photoviewer",

--- a/src/android/PhotoActivity.java
+++ b/src/android/PhotoActivity.java
@@ -1,7 +1,6 @@
 package com.sarriaroman.PhotoViewer;
 
 import android.app.Activity;
-import android.content.Context;
 import android.content.Intent;
 import android.graphics.Bitmap;
 import android.graphics.drawable.BitmapDrawable;
@@ -20,9 +19,9 @@ import android.widget.ProgressBar;
 import android.widget.TextView;
 import android.widget.Toast;
 
+import com.squareup.picasso.Callback;
 import com.squareup.picasso.Picasso;
 import com.squareup.picasso.RequestCreator;
-import com.squareup.picasso.UrlConnectionDownloader;
 
 import org.json.JSONArray;
 import org.json.JSONException;
@@ -32,9 +31,6 @@ import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.lang.reflect.Method;
-import java.net.HttpURLConnection;
-import java.util.Iterator;
-
 import uk.co.senab.photoview.PhotoViewAttacher;
 
 public class PhotoActivity extends Activity {
@@ -199,27 +195,19 @@ public class PhotoActivity extends Activity {
      */
     private void loadImage() throws JSONException {
         if (mImage.startsWith("http") || mImage.startsWith("file")) {
-            Picasso picasso;
-            if (mHeaders == null) {
-                picasso = Picasso.with(PhotoActivity.this);
-            } else {
-                picasso = getImageLoader(this);
-            }
+            this.setOptions(Picasso.get().load(mImage)).into(photo, new Callback() {
+                @Override
+                public void onSuccess() {
+                    hideLoadingAndUpdate();
+                }
 
-            this.setOptions(picasso.load(mImage))
-                    .into(photo, new com.squareup.picasso.Callback() {
-                        @Override
-                        public void onSuccess() {
-                            hideLoadingAndUpdate();
-                        }
+                @Override
+                public void onError(Exception e) {
+                    Toast.makeText(getActivity(), "Error loading image.", Toast.LENGTH_LONG).show();
 
-                        @Override
-                        public void onError() {
-                            Toast.makeText(getActivity(), "Error loading image.", Toast.LENGTH_LONG).show();
-
-                            finish();
-                        }
-                    });
+                    finish();
+                }
+            });
         } else if (mImage.startsWith("data:image")) {
 
             new AsyncTask<Void, Void, File>() {
@@ -231,18 +219,17 @@ public class PhotoActivity extends Activity {
 
                 protected void onPostExecute(File file) {
                     mTempImage = file;
-                    Picasso picasso = Picasso.with(PhotoActivity.this);
 
                     try {
-                        setOptions(picasso.load(mTempImage))
-                                .into(photo, new com.squareup.picasso.Callback() {
+                        setOptions(Picasso.get().load(mTempImage))
+                                .into(photo, new Callback() {
                                     @Override
                                     public void onSuccess() {
                                         hideLoadingAndUpdate();
                                     }
 
                                     @Override
-                                    public void onError() {
+                                    public void onError(Exception e) {
                                         Toast.makeText(getActivity(), "Error loading image.", Toast.LENGTH_LONG).show();
 
                                         finish();
@@ -334,29 +321,5 @@ public class PhotoActivity extends Activity {
             e.printStackTrace();
         }
         return headers;
-    }
-
-    private Picasso getImageLoader(Context ctx) {
-        Picasso.Builder builder = new Picasso.Builder(ctx);
-
-        builder.downloader(new UrlConnectionDownloader(ctx) {
-            @Override
-            protected HttpURLConnection openConnection(Uri uri) throws IOException {
-                HttpURLConnection connection = super.openConnection(uri);
-                Iterator<String> keyIter = mHeaders.keys();
-                String key = null;
-                try {
-                    while (keyIter.hasNext()) {
-                        key = keyIter.next();
-                        connection.setRequestProperty(key, mHeaders.getString(key));
-                    }
-                } catch (JSONException e) {
-                    e.printStackTrace();
-                }
-                return connection;
-            }
-        });
-
-        return builder.build();
     }
 }

--- a/src/android/photoviewer.gradle
+++ b/src/android/photoviewer.gradle
@@ -4,7 +4,7 @@ repositories{
 
 dependencies {
    compile 'com.commit451:PhotoView:1.2.4'
-   compile 'com.squareup.picasso:picasso:2.5.2'
+   compile 'com.squareup.picasso:picasso:2.71828'
 }
 
 android {

--- a/src/ios/PhotoViewer.m
+++ b/src/ios/PhotoViewer.m
@@ -136,7 +136,11 @@
 
 - (NSURL *)localFileURLForImage:(NSString *)image
 {
-    NSString* webStringURL = [image stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLFragmentAllowedCharacterSet]];
+    Boolean isFirebase = [image rangeOfString:@"firebase"].length > 0;
+    NSString* webStringURL = image;
+    if (!isFirebase) {
+      webStringURL = [image stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLFragmentAllowedCharacterSet]];
+    }
     NSURL* fileURL = [NSURL URLWithString:webStringURL];
 
     if (copyToReference && ![fileURL isFileReferenceURL]) {

